### PR TITLE
fix name lookup change again

### LINKED ISSUE #28474: parameter initializer wrongly references variable declared in function body
**TypeScript Version:**  3.2.0-dev.20181101

**Search Terms:**

**Code**

```ts
let baz = 2;
function foo(bar = () => baz) {
    var baz = 1;
    return bar();
}
```

**Expected behavior:**
No error.
`baz` in the initializer references the outer `let baz = 2`.
`let baz = 1` is marked as unused.

**Actual behavior:**
Error `Initializer of parameter 'bar' cannot reference identifier 'baz' declared after it.`

At runtime: calling `foo()` returns `2`, which shows that declarations in the function body are not in scope of parameter initializers.

Note that other parameters **are** in scope and can be used inside other parameters' initializer.

**Playground Link:** https://agentcooper.github.io/typescript-play/#code/DYUwLgBARghgXhAvBATAbgFADMCuA7AYzAEsB7PCLU0gClgCckIaBKJAPmnjYG8MIBEAG4xGsBMgCMmQRHrgc9Cg1aYAvkA

**Related Issues:**
#22769
/cc @Kingwl maybe you want to give this a shot as well (I hope this doesn't cause as much trouble as the last one 😆)

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #28474

### CODE REVIEW

**Comment by Kingwl** [comment]
something seems wrong, i'll fix that later

File: tests/baselines/reference/parameterInitializersForwardReferencing1.errors.txt [hunk 1]
```diff
@@ -57,4 +58,12 @@ tests/cases/conformance/functions/parameterInitializersForwardReferencing1.ts(29
     }
     
     function f8(foo1: string, bar = foo1) { }
+    
+    let bazzzz = 2;
+    function f9(bar = () => bazzzz) {
+                            ~~~~~~
+!!! error TS2373: Initializer of parameter 'bar' cannot reference identifier 'bazzzz' declared after it.
```
Comment (ajafff) [inline tests/baselines/reference/parameterInitializersForwardReferencing1.errors.txt]: why is this error here? It doesn't reference the inner `var bazzzz` (`var` doesn't work any different than `let` and `const` in this case). And the outer `let bazzzz` is definitely not declared after the parameter.

**Comment by Kingwl** [comment]
@ajafff 

```ts
 function bar(func = () => foo) {
     let foo = "in";
     return func
}
```

to 

```ts
function bar(func) {
    if (func === void 0) { func = function () { return foo; }; }
    var foo = "in";
    return func;
}

```

**Comment by Kingwl** [comment]
seems it not a bug? @weswigham

**Comment by weswigham** [comment]
Yeah, I'm pretty sure that error reflects the behavior of our actual down level emit rather than the spec.

**Comment by Kingwl** [comment]
🤷🏻‍♂️ awesome emit

**Pull request closed without merging**